calico/felix/fetcd.py

```diff
@@ -87,20 +90,24 @@
 
 class EtcdAPI(Actor):
     """
-    Our API to etcd.
+    Our API to etcd
 
     Since the python-etcd API is blocking, we defer API watches to
     a worker greenlet and communicate with it via Events.
 
-    As and when we add status reporting, to avoid needing to interrupt
-    in-progress polls, I expect we'll want a second  worker greenlet that
-    manages an "upstream" connection to etcd.
+    To avoid needing to interupt in-progress polls, another working
+    greenlet is introduced to manage status updating and writing to
+    etcd.
     """
 
     def __init__(self, config, hosts_ipset):
         super(EtcdAPI, self).__init__()
         self._config = config
 
+        # Timestamp storing when the EtcdAPI started. This info is needed
+        # in order to report uptime to etcd.
+        self._start_time = time.time()
```

Comment:
`time.time()` shouldn't be used for this; if the clock is changed then we'll misreport our uptime.  We should use a monotonic clock instead.  I made a start on adding such a clock here: https://github.com/projectcalico/calico/compare/monotonic-time